binary_search_tree/tree.py

```diff
@@ -14,35 +14,122 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log(n))
+    # Space Complexity: O(n)
```

Comment:
since you aren't recursively calling this, the space complexity is O(1) because no new memory is dependent upon the size of the input